### PR TITLE
🌱 Upgrade to Go 1.17, controller-runtime v0.11.x, k8s.io v1.23.0-alpha.4

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

This PR upgrades to the latest version of controller-runtime which:
* requires Go 1.17
* uses k8s.io/ v1.23.0-alpha.4

**Depends on: https://github.com/kubernetes/test-infra/pull/24314**

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: controllers/remote/cluster_cache.go [hunk 1]
```diff
@@ -68,7 +68,7 @@ type ClusterCacheTracker struct {
 type ClusterCacheTrackerOptions struct {
 	// Log is the logger used throughout the lifecycle of caches.
 	// Defaults to a no-op logger if it's not set.
-	Log logr.Logger
+	Log *logr.Logger
```
Comment (sbueringer) [inline controllers/remote/cluster_cache.go:71]: Note: logr.Logger is now a struct not an interface. I made it a pointer to be able to check if the option is set.

I think there is simply no way to keep the API stable when go-logr changes the Logger interface to a struct (from v0.4=>v1.x)

**Comment by sbueringer** [comment]
/hold
for:
* CR/CT releases (currently go.mod is using their latest commits on main)
* https://github.com/kubernetes/test-infra/pull/24314 must be merged first

**Review by vincepri — APPROVED** [review]
/approve
/lgtm
/hold cancel

**Comment by vincepri** [comment]
/retest

Another try

**Comment by sbueringer** [comment]
I'll take a look

File: .github/workflows/dependabot.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ jobs:
     - name: Set up Go 1.x
       uses: actions/setup-go@v2
       with:
-        go-version: '1.16'
+        go-version: '1.17'
```
Comment (fabriziopandini) [inline .github/workflows/dependabot.yml:20]: Q: are we still using dependant or can we drop this file entirely (not blocking for this PR)

Comment (sbueringer) [reply .github/workflows/dependabot.yml:20]: Afaik it's currently not running. I can drop it in a follow-up PR when we have consensus to drop it.
(/cc @randomvariable )

**Comment by sbueringer** [comment]
full-e2e tests are essentially also green, the failed one will be fixed in https://github.com/kubernetes-sigs/cluster-api/pull/5625#pullrequestreview-803971032 and is independent of this PR

**Pull request merged**